hips/hip-XXXX.md

```diff
@@ -0,0 +1,157 @@
+Today, Helm will only install CRDs (from the `crds/` directory) when installing a chart.
+CRDs are otherwise ignored for additional operations, including upgrades.
+
+CRDs have become more commonly utilized to extend Kubernetes functionality beyond its core primatives.
+Being conservative ensured Helm erred on the side of safety and didn't create an irreversible position for Helm nor its users.
+But has limited Helm's users, both chart operators and authors, to produce charts that expect to be able to manage CRDs.
+As such, improvements to Helm's CRD management, that allows Helm and charts to better manage and update CRDs are desired.
+
+The solution here aims to extend Helm's CRD management, while adhering to the concerns detailed in [HIP-0011](./hip-0011.md).
+While providing for better CRD management, but continuing to provide for safe operation of charts by chart operators.
+
+
+## Rationale
+
+<!--
+Describe why particular design decisions were made.
+-->
+The two main reasons inferred from [HIP-0011](./hip-0011.md) that Helm is conservative with CRDs are:
+1. CRDs are a cluster-wide resource.
+   Changes to cluster wide resources can (more easily) break applications beyond the scope of the chart
+2. Custom Resources (CRs) can be used as a data store.
+   And removing the CRDs for those will irrevocably remove any CRs and cause parmenat loss of any contained data
+
+For 1., it is thought that Helm should not treat CRDs specially here.
+Helm will readily operate on many other cluster wide resources today: Cluster roles, priority classes, namespaces, etc.
+That the modification/removal of could easily cause breakage outside of the chart's release.
+
+In general, Helm as a package manager should not try to prempt unintended functional changes from a chart.
+Validating functional changes is well beyond the scope of a package manager.
+Helm should treat CRDs the same as other (cluster-wide) resources, where a Helm chart upgrade that causes unintended functional effects should be reverted (rolled back) by the user (chart operator).
+And as long as that rollback path exists, this is a suitable path for users to mitigate breaking functional changes.
+
+For 2., Data loss should be treated more carefully.
+As data loss can be irrevocable or require significant effort to restore.
+And especially, an application/chart operator should not expect a chart upgrade to cause data loss.
+Helm can prevent Data loss can be prevented by ensuring CRDs are "appended" only (with special exceptions in the specification below). An in particular, appending allows a rollback to (effectively) restore existing cluster state.
+(It should also be noted that Helm will today remove other resources which may cause data loss e.g. secrets, config maps, namespace, etc. A special, hard-coded, exception does exist for persistent volumes)
```

Comment:
Data loss is different between CRDs and something like secrets. If I remove a secret it removes the one secret. This impacts that single case. If a CRD is deleted all the CRs are deleted. For example, you have user A with application instance of a chart. And you have user B in that same cluster with a separate instance of the same chart. These 2 users do not know about each other. If user A deletes a CRD, even unintentionally, it will remove the CRs for user B and cause data loss for user B. This is a very different surface area from something like deleting a secret.

We also know that some don't have backups and will make changes in production. When things go wrong, Helm devs will get issues to triage and deal with.

Reply:
I should probably delete this last sentence, I'm not sure why I added it. The explicit intention of this HIP is to ensure Helm can not cause data loss with CRDs

Reply:
(removed)